tests/child_chain/test_child_chain.py

```diff
@@ -53,22 +58,72 @@ def test_send_tx_invalid_sig(child_chain):
     tx3 = Transaction(1, 0, 0, 1, 1, 0, newowner1, amount2, b'\x00' * 20, 0, 0)
 
     # Sign with an invalid key
-    tx3.sign1(invalid_key)
-    tx3.sign2(invalid_key)
+    tx3.sign1(invalid_tx_key)
+    tx3.sign2(invalid_tx_key)
 
     with pytest.raises(AssertionError):
         child_chain.apply_transaction(rlp.encode(tx3).hex())
 
 
-def test_send_tx_double_spend(child_chain, u):
+def test_send_tx_double_spend(child_chain):
     tx3 = Transaction(1, 0, 0, 1, 1, 0, newowner1, amount2, b'\x00' * 20, 0, 0)
 
-    tx3.sign1(key1)
-    tx3.sign2(key1)
+    tx3.sign1(tx_key)
+    tx3.sign2(tx_key)
 
     # Submit once
     child_chain.apply_transaction(rlp.encode(tx3).hex())
 
     # Try to submit again
     with pytest.raises(AssertionError):
         child_chain.apply_transaction(rlp.encode(tx3).hex())
+
+
+def test_submit_block(child_chain):
+    block = child_chain.current_block
+    block.make_mutable()
+    block.sign(block_key)
+    block = rlp.encode(block).hex()
+
+    old_block_number = child_chain.current_block_number
+    child_chain.submit_block(block)
+    assert child_chain.current_block_number == old_block_number + 1
+
+
+def test_submit_block_no_sig(child_chain):
+    block = child_chain.current_block
+    block = rlp.encode(block).hex()
+
+    with pytest.raises(AssertionError):
+        child_chain.submit_block(block)
+
+
+def test_submit_block_invalid_sig(child_chain):
+    block = child_chain.current_block
+    block.make_mutable()
+    block.sign(invalid_block_key)
+    block = rlp.encode(block).hex()
+
+    with pytest.raises(AssertionError):
```

Comment:
Minor: I would recommend to assert some message as well in submit_block
eg.
```python
assert valid_signature, 'invalid signature'
```

then here can be
```python
with pytest.raises(AssertionError) as e:
    child_chain.submit_block(block)
assert str(e.value) == 'invalid signature'
```
So generally we can make sure we are testing on the assert that we really want, and not from other places. (Though probably not the case here)

Or, personally I prefer to have custom Exceptions rather than assert. But it seems like the package have the convention of using assert, so I am not sure moving to custom Exception would be better or not.

Reply:
I don't think we're set on the usage of `AssertionError`, I agree that it's probably better to throw custom errors. Maybe something like `InvalidBlockSignatureError`? Is that too wordy? It's relatively easy to implement these as:
```
class InvalidBlockSignatureError(Exception):
    """Raise when a block is signed by anyone but the operator"""
```